### PR TITLE
Change order updater not to save

### PR DESCRIPTION
**Pull request by mbj**
Extraction from #398, we do need more control about when `#shipment_total` is written.

Depends on #491

### CODE REVIEW

File: core/app/models/spree/order_updater.rb [hunk 1]
```diff
@@ -11,7 +11,6 @@ def update
       update_payment_state
       update_shipments
       update_totals
-      order.save
```
Comment (dkubb) [inline core/app/models/spree/order_updater.rb]: Are there other callers of this method besides the ones in this diff? If so then maybe they should explicitly save, and we can then look at each one to remove it after we've verified it is handled elsewhere.

Comment (mbj) [inline core/app/models/spree/order_updater.rb]: There are the following cases where order states are written too:

* Order populator from `OrdersController#populate` (save added this PR to the `OrderContents` that will be merged to the controller at some point)
* Order update from `OrdersController#update` (backed by `OrderContents` also)
* Admin adjustments controller, explicit save added
* Checkout, save triggered from state machine, and shipment stuff (that one should be removed)
* Admin payment captures, I still need to validate, good catch.
* Admin cancelation / resume, I still need to validate, good catch

**Comment by mbj** [comment]
@dkubb I tested the additional scenarios with success, `#save` call sides exit for all scenarios.

**Comment by mbj** [comment]
@dkubb This is ready for review again.

**Pull request merged**